scheduler/context.go

```diff
@@ -124,3 +129,121 @@ func (e *EvalContext) ProposedAllocs(nodeID string) ([]*structs.Allocation, erro
 	}
 	return proposed, nil
 }
+
+func (e *EvalContext) Eligibility() *EvalEligibility {
+	if e.eligibility == nil {
+		e.eligibility = NewEvalEligibility()
+	}
+
+	return e.eligibility
+}
+
+type ComputedClassEligibility byte
+
+const (
+	// The EvalComputedClass enums denote the eligibility of the computed class
+	// for the evaluation.
+	EvalComputedClassUnknown ComputedClassEligibility = iota
+	EvalComputedClassIneligible
+	EvalComputedClassEligible
+	EvalComputedClassEscaped
```

Comment:
I'd add more docs for each value to describe the implications, especially the `Escaped` one is non-obvious.